src/libostree/ostree-repo-static-delta-core.c

```diff
@@ -409,7 +409,15 @@ ostree_repo_static_delta_execute_offline_with_signature (OstreeRepo *self, GFile
         return glnx_throw_errno_prefix (error, "openat(O_DIRECTORY)");
       else
         {
-          g_autofree char *dir = dirname (g_strdup (dir_or_file_path));
+          g_autofree char *dir = "";
+          if (dir_or_file_path[strlen (dir_or_file_path) - 1] == G_DIR_SEPARATOR)
```

Comment:
Hold on, now our logic is getting more complicated here...why the conditional? Is this because `g_path_get_dirname()` has different semantics than `dirname`?

Reply:
> Is this because g_path_get_dirname() has different semantics than dirname?

Yeah. `dirname` will ignore `/` at the end of a filepath while `g_path_get_dirname` will not. For example, `dirname("test/path/example/")` will return `"test/path"` while `g_path_get_dirname("test/path/example/")` will return `test/path/example`